docs/conf.py

```diff
@@ -169,8 +169,11 @@
 
 # -- Turn on nitpicky mode for sphinx (to warn about references not found) ----
 #
-# nitpicky = True
-# nitpick_ignore = []
+nitpicky = True
+nitpick_ignore = [
```

Comment:
ah, thanks for that tip. updating `intersphinx_mapping` is the trick that i was looking for, but could not easily find. 

sphinx was resolving the class `specutils.Spectrum1D` to `specutils.spectra.spectrum1d.Spectrum1D` in `automodapi` which is why i did the `py:class` ignore for the latter. when mentioned in regular `rst` docs it did not resolve so i had the `py:obj` ignore for that case. anyway, manually updating the `intersphinx_mapping` does take care of this, but there's still something janky happening to cause the mapping to not work out of the box.

Reply:
intersphinx is only as good as the upstream doc you are using... Could it be some API doc has not made it into a release or has been removed and so on? What is not working for you?

Reply:
what wasn't working was that `specutils.Spectrum1D` does not map automatically whereas classes from `astropy`, `numpy`, and everything else do. that's not just some API doc, that's the basic class for that package. it's imported in that file and it is in the build's environment. sphinx can even figure out where the code for `specutils.Spectrum1D` actually lives within the package. however, it's somehow not getting the link to the docs from all that.

otoh, i manually added `ccdproc` because it is not a requirement (yet) and is not available in the `build_docs` environment. that's totally fair.

Reply:
https://github.com/spacetelescope/jdaviz/blob/78b5d3584784a3e9cf134b506935e9e79d872064/docs/conf.py#L267

https://jdaviz.readthedocs.io/en/latest/specviz/import_data.html

We used this at Jdaviz and it renders fine. Hmm.

```rst
:class:`~specutils.Spectrum1D`
```

Reply:
it renders fine once i update `intersphinx_mapping`. does jdaviz do that?

Reply:
Yes, Jdaviz uses intersphinx. If it renders after intersphinx is defined, then what is the problem? I am a little confused.

Reply:
Ahhh... yes, astropy and numpy are already defined for you upstream here:

https://github.com/astropy/sphinx-astropy/blob/6928e75f53f3988e55121da5cd485fadfb2d8455/sphinx_astropy/conf/v1.py#L54

Anything else, you have to add it yourself. Does that clarify the matter?

Reply:
ah yes! that does clarify things greatly. and i see in https://github.com/astropy/sphinx-astropy/issues/9 a 4 year old conversion whether to expand or contract the list of packages mapped by default. i'll take things up there.